### PR TITLE
Ambiguity in function overloading in c++

### LINKED ISSUE #2362: CPP: Ambiguity in Function Overloading
### Description

I will provide the detailed documentation on ambiguity in function overloading in c++.

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by ShikhaSwati60**
<hr>

## Description 📜

Described the cases in which function overloading can occur.
Fixes #2362

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by ShikhaSwati60** [comment]
@Saviour1001 please check my pr

**Comment by ChetasShree** [comment]
Amazing work @ShikhaSwati60 !!
@Saviour1001 @Suvraneel pls review this pr and merge it :)
Thank you !!

File: C_CPP/OOPs in C++/Polymorphism or Overloading/readme.md [hunk 1]
```diff
@@ -1 +1 @@
-
+* Ambiguity in Function Overloading
```
Comment (Saviour1001) [inline C_CPP/OOPs in C++/Polymorphism or Overloading/readme.md:1]: Add a relative link to the file here. Here is how you add a [relative link] (https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file)

Comment (ShikhaSwati60) [reply C_CPP/OOPs in C++/Polymorphism or Overloading/readme.md]: @Saviour1001 I have made the changes.

File: C_CPP/OOPs in C++/Polymorphism or Overloading/readme.md [hunk 2]
```diff
@@ -1 +1 @@
-* Ambiguity in Function Overloading
+[Ambiguity in Function Overloading in C++](Ambiguity_in_Function_Overloading.md)
```
Comment (ChetasShree) [inline C_CPP/OOPs in C++/Polymorphism or Overloading/readme.md:1]: add a '-' 
- [Ambiguity in Function Overloading in C++](Ambiguity_in_Function_Overloading.md)
type like this

**Review by ChetasShree — CHANGES_REQUESTED** [review]

File: C_CPP/OOPs in C++/Polymorphism or Overloading/readme.md (hunk 2 above)
Comment (ShikhaSwati60) [reply C_CPP/OOPs in C++/Polymorphism or Overloading/readme.md]: @ChetasShree please check it once

**Comment by ChetasShree** [comment]
@Saviour1001 @Suvraneel check it ones

**Pull request merged**